### PR TITLE
Problem: client can't create deposit bonded stake, unbond stake and withdraw unbonded stake transactions (CRO-174)

### PR DESCRIPTION
**Pull request by leejw51crypto**
add two functions to DefaultNetworkOpsClient

---
This change is [Reviewable](https://reviewable.io/reviews/crypto-com/chain/154)

### CODE REVIEW

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 1]
```diff
@@ -32,6 +48,86 @@ impl<'a, W> NetworkOpsClient for DefaultNetworkOpsClient<'a, W>
...
+        from_address: &ExtendedAddr,
+        value: Coin,
+        attributes: StakedStateOpAttributes,
+    ) -> Result<TxAux> {
+        match from_address {
+            ExtendedAddr::BasicRedeem(ref redeem_address) => {
+                let secp = Secp256k1::new();
+                let transaction: UnbondTx = UnbondTx {
+                    value,
+                    nonce: 0,
```
Comment (devashishdxt) [inline client-network/src/network_ops/default_network_ops_client.rs]: I don't know what `nonce` should we set here. For now, `0` is ok but probably add a `TODO` to change it later when we have more clarity. Any comments @tomtau?

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 2]
```diff
@@ -32,6 +48,86 @@ impl<'a, W> NetworkOpsClient for DefaultNetworkOpsClient<'a, W>
...
+                let public_key = self
+                    .wallet_client
+                    .find_public_key(name, passphrase, redeem_address)?
+                    .ok_or_else(|| Error::from(ErrorKind::AddressNotFound))?;
+
+                let private_key = self
+                    .wallet_client
+                    .private_key(passphrase, &public_key)?
+                    .ok_or_else(|| Error::from(ErrorKind::PrivateKeyNotFound))?;
+                let secret_key: SecretKey = SecretKey::from(&private_key);
```
Comment (devashishdxt) [inline client-network/src/network_ops/default_network_ops_client.rs]: Signing should be done using `sign()` function on `private_key`.

Something similar to this:
```
let signature = private_key
    .sign(transaction.id())
    .map(StakedStateOpWitness::new)?;
```

File: client-network/Cargo.toml [hunk 3]
```diff
@@ -12,3 +12,5 @@ chain-tx-validation = { path = "../chain-tx-validation" }
 client-common = { path = "../client-common" }
 client-core = { path = "../client-core" }
 secstr = "0.3.2"
+secp256k1zkp = { git = "https://github.com/crypto-com/rust-secp256k1-zkp.git", rev = "a1e70e1f0bb65d3a97f100f79d99c85395d897f8", features = ["recovery", "endomorphism"] }
```
Comment (devashishdxt) [inline client-network/Cargo.toml]: I don't think we should add this dependency in here. Unless we explicitly need to use some of its functionality in this crate.

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 4]
```diff
@@ -32,6 +48,86 @@ impl<'a, W> NetworkOpsClient for DefaultNetworkOpsClient<'a, W>
...
+                let witness2 = vec![TxInWitness::BasicRedeem(get_ecdsa_witness(
```
[the remark this answers is not in the document]
Comment (tomtau) [reply client-network/src/network_ops/default_network_ops_client.rs]: (replied below using reviewable.io)

File: client-network/src/network_ops/default_network_ops_client.rs (hunk 1 above)
Comment (tomtau) [reply client-network/src/network_ops/default_network_ops_client.rs]: (replied below using reviewable.io)

**Comment by leejw51crypto** [comment]
i fixed above issues.

File: client-network/Cargo.toml (hunk 3 above)
Comment (devashishdxt) [inline client-network/Cargo.toml]:
```suggestion
```

File: client-network/src/network_ops.rs [hunk 5]
```diff
@@ -23,5 +43,5 @@ pub trait NetworkOpsClient {
         attributes: TxAttributes,
     ) -> Result<TxAux>;
 
-    // TODO: Add `create_deposit_bonded_stake_transaction()`, `create_unbond_stake_transaction()` and `get_account_details()` functions
+    // TODO: Add `get_account_details()` functions
```
Comment (devashishdxt) [inline client-network/src/network_ops.rs]: Now that we have changed name from `Account` to `StakedState`
```suggestion
    // TODO: Add `get_staked_state()` functions
```

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 6]
```diff
@@ -1,37 +1,105 @@
...
+        &self,
+        name: &str,
+        passphrase: &SecUtf8,
+        from_address: &ExtendedAddr,
+        value: Coin,
+        attributes: StakedStateOpAttributes,
+    ) -> Result<TxAux> {
+        match from_address {
+            ExtendedAddr::BasicRedeem(ref redeem_address) => {
+                let transaction = UnbondTx::new(value, 0, attributes);
```
Comment (devashishdxt) [inline client-network/src/network_ops/default_network_ops_client.rs]: Add a `TODO` to get `nonce` from `StakedState`.

File: client-network/src/network_ops/default_network_ops_client.rs [hunk 7]
```diff
@@ -42,26 +110,18 @@ where
     ) -> Result<TxAux> {
         match from_address {
             ExtendedAddr::BasicRedeem(ref redeem_address) => {
-                let transaction = WithdrawUnbondedTx {
-                    nonce: 0,
-                    outputs,
-                    attributes,
-                };
-
+                let transaction = WithdrawUnbondedTx::new(0, outputs, attributes);
```
Comment (devashishdxt) [inline client-network/src/network_ops/default_network_ops_client.rs]: Add a `TODO` to get `nonce` from `StakedState`.

File: client-network/Cargo.toml (hunk 3 above)
Comment (leejw51crypto) [reply client-network/Cargo.toml]: agree

**Pull request merged**